cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -329,14 +400,24 @@ class HeffteFastFourierTransform
     using device_type = DeviceType;
     using backend_type = BackendType;
     using exec_space = typename device_type::execution_space;
+    using mesh_type = MeshType;
+    static constexpr std::size_t num_space_dim = mesh_type::num_space_dim;
+
     using heffte_backend_type =
         typename Impl::HeffteBackendTraits<exec_space,
                                            backend_type>::backend_type;
+    using heffte_fft_type = std::conditional_t<
+        3 == num_space_dim, heffte::fft3d<heffte_backend_type>,
+        std::conditional_t<2 == num_space_dim,
+                           heffte::fft2d<heffte_backend_type>, void>>;
+    using heffte_box_type = std::conditional_t<
+        3 == num_space_dim, heffte::box3d,
+        std::conditional_t<2 == num_space_dim, heffte::box2d, void>>;
```

Comment:
These may be unnecessary. heFFTe's `fft2d` and `box2d` are just aliases for `fft3d` and `box3d`. What I understand is that heFFTe should be able to detect geometry and act appropriately.

Reply:
Good point - I simplified and commented that 3d can also handle 2d